src/core.rs

```diff
@@ -92,6 +99,7 @@ impl Core {
                 Ok(meta) => meta,
                 Err(err) => {
                     print_error!("{}: {}.", path.display(), err);
+                    self.exit_code.set_if_greater(ExitCode::MajorIssue);
```

Comment:
Instead of having exit code be a field in `Core` and pass into fetch. It can just be a local value that is initialized here and then returned back to the run function similar to what you have done for `recurse_into`.